src/test/suite/soap_messages.test.ts

```diff
@@ -0,0 +1,110 @@
+        const zip = new AdmZip();
+        filenameMapping.forEach((pathInRepo, nameInZip) => {
+            zip.addLocalFile(pathInRepo, '', nameInZip);
+        })
+
+        finalResponse = Buffer.concat([
+            Buffer.from(boundary + '\r\n'),
+            Buffer.from(firstPart + '\r\n'),
+            Buffer.from(boundary + '\r\n'),
+            Buffer.from(headerLines.join('\r\n') + '\r\n\r\n'),
+            zip.toBuffer(),
+            Buffer.from('\r\n' + boundary + '--')
+        ]);
+
+    })
+
+    test('Extract server logs correctly', () => {
+        // Check that we extract the same logs we put in
+        const logs = extractLogs(finalResponse);
+        filenameMapping.forEach((pathInRepo, nameInZip) => {
+            assert.strictEqual(
+                logs.get(nameInZip),
+                fs.readFileSync(pathInRepo).toString()
+            );
+        });
+    })
+
+    test('Generate initial message correctly', () => {
+        // Create the initial request to compare against the expected one
+        const filename = "belsunu.atf"
+        const text = fs.readFileSync(path.join(inputPath, filename)).toString();
+        const msg = (new MultipartMessage("atf", filename, "cams/gkab", text))._message;
+        const expected = fs.readFileSync(path.join(refPath, 'initial_request_belsunu')).toString();
+        // We can't directly compare the results, because the date is encoded
+        // in the zip files, so those will differ. We can compare them in parts though.
+        // The boundary is the first 7 characters
+        const expectedParts = expected.split('--uafdHS3v');
+        // Compare the first part (SOAP envelope)
+        const expectedSOAP = expectedParts[1]
+                                // skip the first and last line change (not included in message)
+                                .trim()
+                                // ignore how line endings are encoded
+                                .replace(/\r\n/g, "\n")
+        assert.strictEqual(msg._body[0].toString().replace(/\r\n/g, "\n"),
+                           expectedSOAP);
+        // Get and compare the headers for the second part (zip attachment)
+        const attachmentLines: string[] = msg._body[1].toString().split('\r\n');
+        const numHeaders = 4;
+        const attachmentHeaders = attachmentLines.slice(0, numHeaders);
+        const expectedHeaders = expectedParts[2].trim().split(/\r?\n/).slice(0, numHeaders);
+        // Need "deep" equality to compare array contents
+        assert.deepStrictEqual(attachmentHeaders, expectedHeaders);
+        // Get the zip contents, read as a zip file and compare with original text
+        // There should only be one line but just in case \n happens to be in the encoded zip...
+        const zipContents = Buffer.from(attachmentLines.slice(numHeaders + 1).join(""),
+                                        'latin1');
+        fs.writeFileSync('test_zip_buffer', zipContents);
```

Comment:
After running the tests I have a file called `test_zip_buffer` in my current directory, this should be cleaned up